.github/workflows/ci.yaml

```diff
@@ -0,0 +1,26 @@
+---
+ name: ci
+ on:
+   - push
+ jobs:
+   lint:
+     name: lint
+     runs-on: ubuntu-latest
+     steps:
+       - uses: actions/checkout@v2
+       - name: golangci-lint
+         uses: golangci/golangci-lint-action@v2
+         with:
+           version: latest
+   test:
+     name: test
+     runs-on: ubuntu-latest
+     steps:
+       - uses: actions/checkout@v2
+       - uses: actions/setup-go@v2
+         with:
+           go-version: 1.22
+       - run: go test -v ./... -covermode=atomic -coverprofile=coverage.out
+       - uses: codecov/codecov-action@v1
```

Comment:
I think you need to consider adding a codecov token for it functioning properly. Also, you can use [Shields](https://shields.io/) to add badges 🙈